helper/memory_calculator.go

```diff
@@ -220,3 +227,25 @@ func parseMemInfo(s string) (int64, error) {
 	}
 	return num * unit, nil
 }
+
+func CountAgentClasses(opts string) (int, error) {
+	var agentClassCount int
+	if p, err := shellwords.Parse(opts); err != nil {
+		return 0, fmt.Errorf("unable to parse $JAVA_TOOL_OPTIONS\n%w", err)
+	} else {
+		var agentPaths []string
+		for _, s := range p {
+			if strings.HasPrefix(s, "-javaagent:") {
+				agentPaths = append(agentPaths, strings.Split(s, ":")[1])
+			}
+		}
+		for _, path := range agentPaths {
+			if c, err := count.Classes(filepath.Dir(path)); err == nil {
+				agentClassCount += c
+			} else {
+				return 0, fmt.Errorf("unable to count classes of agent jar at %s\n%w", path, err)
+			}
+		}
```

Comment:
Two suggestions here:

1. I think you can safely call `count.JarClasses` instead of `count.Classes`. The "module" referenced there is a Java 9+ module and I don't think anything outside of the JVM is using those. Plus, I think `-javaagent:` expects either a JAR so again, we should be Ok with directly calling `JarClasses`.

2. You might move this loop into a method under `count/count_classes.go`, maybe `func JarClassesFrom(path ...string) (int, error)`, which just iterates and calls `JarClasses` multiple times. It would shorten up this method a bit and make it more reusable.